src/reflect/scala/reflect/internal/Symbols.scala

```diff
@@ -1535,14 +1535,22 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
 //          activeLocks += 1
  //         lockedSyms += this
         }
-        val current = phase
-        try {
-          assertCorrectThread()
-          phase = phaseOf(infos.validFrom)
-          tp.complete(this)
-        } finally {
-          unlock()
-          phase = current
+        if (isCompilerUniverse) {
+          val current = phase
+          try {
+            assertCorrectThread()
+            phase = phaseOf(infos.validFrom)
+            tp.complete(this)
+          } finally {
+            unlock()
+            phase = current
+          }
+        } else {
```

Comment:
This could use a short comment to explain why we don't (need to) adjust the phase to the `validFrom` phase for this symbol's type history.